components/add_users_to_team_modal/add_users_to_team_modal.tsx

```diff
@@ -0,0 +1,254 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {Modal} from 'react-bootstrap';
+import {FormattedMessage} from 'react-intl';
+
+import {UserProfile} from 'mattermost-redux/types/users';
+import {Team} from 'mattermost-redux/types/teams';
+
+import {Client4} from 'mattermost-redux/client';
+
+import {displayEntireNameForUser, localizeMessage, isGuest} from 'utils/utils.jsx';
+import ProfilePicture from 'components/profile_picture';
+import GuestBadge from 'components/widgets/badges/guest_badge';
+import BotBadge from 'components/widgets/badges/bot_badge';
+import MultiSelect, {Value} from 'components/multiselect/multiselect';
+import AddIcon from 'components/widgets/icons/fa_add_icon';
+
+const USERS_PER_PAGE = 50;
+const MAX_SELECTABLE_VALUES = 20;
+
+type UserProfileValue = Value & UserProfile;
+
+type Props = {
+    team: Team;
+    users: UserProfile[];
+    excludeUsers: { [userId: string]: UserProfile };
+    includeUsers: { [userId: string]: UserProfile };
+    onAddCallback: (users: UserProfile[]) => void;
+    onHide?: () => void;
+
+    actions: {
+        getProfilesNotInTeam: (teamId: string, groupConstrained: boolean, page: number, perPage?: number) => Promise<{ data: UserProfile[] }>;
+        searchProfiles: (term: string, options?: any) => Promise<{ data: UserProfile[] }>;
+    };
+}
+
+type State = {
+    searchResults: UserProfile[];
+    values: UserProfileValue[];
+    show: boolean;
+    search: boolean;
+    saving: boolean;
+    addError: null;
+    loading: boolean;
+}
+
+export default class AddUsersToTeamModal extends React.PureComponent<Props, State> {
+    public constructor(props: Props) {
+        super(props);
+
+        this.state = {
+            searchResults: [],
+            values: [],
+            show: true,
+            search: false,
+            saving: false,
+            addError: null,
+            loading: true,
+        };
+    }
+    public componentDidMount = async () => {
+        await this.props.actions.getProfilesNotInTeam(this.props.team.id, false, 0, USERS_PER_PAGE * 2);
+        this.setUsersLoadingState(false);
+    }
+
+    private setUsersLoadingState = (loading: boolean) => {
+        this.setState({loading});
+    }
+
+    public search = async (term: string) => {
+        this.setState({loading: true});
```

Comment:
I can - but the purpose of that function is not really to avoid having to write out `this.setState...` its more cause the `componentDidMount` function does not have access to `this.setState` so we have to create a function to call from there to set the state.

Reply:
Ah okay. Why don't you create a general function than something like `setLoadingState` so it can be used in other places rather than in just one place?